src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -90,17 +90,55 @@
     return future;
   }
 
+  public CompletableFuture<InvoiceLine> getInvoiceLinePersistTotal(String id) {
+    CompletableFuture<InvoiceLine> future = new VertxCompletableFuture<>(ctx);
+
+    try {
+      // 1. GET invoice-line from storage
+      getInvoiceLine(id).thenAccept(invoiceLineFromStorage -> {
+        logger.info("Successfully retrieved invoice line: " + invoiceLineFromStorage);
+
+        // 2. Save invoice-line total from storage for future comparison
+        Double existingTotal = invoiceLineFromStorage.getTotal();
+
+        // 3. Calculate invoice-line totals, if different from storage, write it back to storage
+        calculateInvoiceLineTotals(invoiceLineFromStorage).thenAccept(invoiceLineWithTotalRecalculated -> {
+          Double recalculatedTotal = invoiceLineWithTotalRecalculated.getTotal();
+          if (Double.compare(recalculatedTotal, existingTotal) != 0) {
+            updateInvoiceLineToStorage(invoiceLineWithTotalRecalculated)
+              .thenAccept(updateSuccess -> future.complete(invoiceLineWithTotalRecalculated));
+          } else {
+            future.complete(invoiceLineFromStorage);
+          }
+        });
+      })
+        .exceptionally(t -> {
+          logger.error("Error getting invoice line by id ", id);
```

Comment:
Looks like this `exceptionally` is used just to log an error message. But the `getInvoiceLine()` method already logs a similar message. Do we really need to repeat the message?

Reply:
I have updated log messages to be more specific. It makes sense to log an error message if it fails to persist total. I think it will show the cause of failure right away and will help in troubleshooting.

Reply:
This exceptionally is currently processing exceptions only from `getInvoiceLine()`.